### ISSUE TITLE
Alternator doesn't mix well with TWCS, either document or disallow it explicitly

### ISSUE THREAD

**Issue by raphaelsc** [scylladb/scylladb#19805]
Quoting me

```
TWCS doesn't handle tombstones well, as we don't compact across
windows. So tombstone can sit in window W1, whereas shadowed data in
window W2. And if I understand Alternator correctly, the sstable TTL
is not used for it. So we could find ourselves in a situation where
nothing ever gets removed from disk.
```

Quoting @nyh

```
Wow, good catch :-(

Indeed, whereas with CQL TTL, TWCS can say that an entire old time window is
entirely expired and just throw it away, with the Alternator TTL feature the expiration
events look like tombstones in a *new* time window, and TWCS will never consider
the old and new time window together, and nothing will ever get deleted. So TWCS
is indeed a bad match for Alternator TTL feature.
```

we either need to document it or disallow it if user decides to perform an alter table (via CQL), if that's possible.

**Comment by mykaul** [comment]
I prefer disallowing if it makes no sense - and I don't see under which conditions it does.

**Comment by dorlaor** [comment]
How can someone even define TWCS with the alternator? It should default to
ICS w/o an
ability to change (other than hacking)

**Comment by tzach** [comment]
> ICS w/o an ability to change (other than hacking)

Exactly. @raphaelsc, is this a theoretical risk, or did someone use CQL to ALTER an Alternator table?

**Comment by raphaelsc** [comment]
> > ICS w/o an ability to change (other than hacking)
> 
> Exactly. @raphaelsc, is this a theoretical risk, or did someone use CQL to ALTER an Alternator table?

Theoretical. Do we expose cql for alternator users?

**Comment by ptrsmrn** [comment]
> Theoretical. Do we expose cql for alternator users?

Authorization for Alternator will be accomplished by CQL, see https://github.com/scylladb/scylladb/pull/19762

**Comment by nyh** [comment]
> Theoretical. Do we expose cql for alternator users?

We didn't want to originally, but we may have to (as @ptrsmrn said, for authorization).
However, it doesn't mean that users are just supposed to use random CQL commands. They can break their data in all sorts of ways with CQL, I'm not sure we need to document all the ways to break the data by using CQL commands on the Alternator database.

I think it won't hurt to document this fact (that TWCS should be used for CQL TTL but not Alternator TTL), but looking at https://opensource.docs.scylladb.com/stable/cql/compaction.html - it contains hardly any explanation on when to use or not use TWCS, so I'm not even sure where to add such documentation.
